paper/src/main/java/com/denizenscript/denizen/paper/properties/PaperItemExtensions.java

```diff
@@ -0,0 +1,57 @@
+package com.denizenscript.denizen.paper.properties;
+
+import com.denizenscript.denizen.nms.NMSHandler;
+import com.denizenscript.denizen.nms.NMSVersion;
+import com.denizenscript.denizen.objects.EntityTag;
+import com.denizenscript.denizen.objects.ItemTag;
+import com.denizenscript.denizencore.objects.core.ElementTag;
+import com.denizenscript.denizencore.objects.core.MapTag;
+
+public class PaperItemExtensions {
+
+    public static void register() {
+
+        // <--[tag]
+        // @attribute <ItemTag.rarity>
+        // @returns ElementTag
+        // @group properties
+        // @Plugin Paper
+        // @description
+        // Returns the rarity of an item, as "common", "uncommon", "rare", or "epic".
+        // -->
+        ItemTag.tagProcessor.registerTag(ElementTag.class, "rarity", (attribute, item) -> {
+            return new ElementTag(item.getItemStack().getRarity());
+        });
+
+        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_19)) {
+
+            // <--[mechanism]
+            // @object EntityTag
+            // @name damage_item
+            // @input MapTag
+            // @Plugin Paper
+            // @description
+            // Damages the ItemTag in the given entities' inventory for the given amount.

```

Comment:
Is being in the entity's inventory required? seems like it should work with any `ItemStack`, and just needs an entity to base the logic on & apply stat changes and whatnot to.